File: Leetcode-Problems/Evaluate Reverse Polish Notation.cpp
class Solution
{
public:
    int evalRPN(vector<string> &tokens)
    {
        stack<int> ans;
        int temp;
        for (string s : tokens)
        {
            if (s == "+" || s == "-" || s == "*" || s == "/")
            {
                int op1 = ans.top();
                ans.pop();
                int op2 = ans.top();
                ans.pop();
                if (s == "+")
                {
                    temp = op2 + op1;
                }
                else if (s == "-")
                {
                    temp = op2 - op1;
                }
                else if (s == "*")
                {
                    temp = op2 * op1;
                }
                else if (s == "/")
                {
                    temp = op2 / op1;
                }
                ans.push(temp);
            }
            else
            {
                ans.push(stoi(s));
            }
        }
        return ans.top();
    }
};

File: Leetcode-Problems/Evaluate Reverse Polish Notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> st = new Stack<>();
        for (String s : tokens) {
            if (st.size() > 1 && s.equals("+")) {
                int a = st.pop();
                int b = st.pop();
                st.push(a + b);
            } else if (st.size() > 1 && s.equals("-")) {
                int a = st.pop();
                int b = st.pop();
                st.push(b - a);
            } else if (st.size() > 1 && s.equals("*")) {
                int a = st.pop();
                int b = st.pop();
                st.push(a * b);
            } else if (st.size() > 1 && s.equals("/")) {
                int a = st.pop();
                int b = st.pop();
                st.push(b / a);
            } else {
                st.push(Integer.parseInt(s));
            }
        }
        return st.peek();
    }
}